server/src/main/java/org/elasticsearch/search/SearchService.java

```diff
@@ -641,6 +642,9 @@ private DefaultSearchContext createSearchContext(ShardSearchRequest request, Tim
         } finally {
             if (success == false) {
                 IOUtils.closeWhileHandlingException(searchContext);
+                if (searchContext == null) {
```

Comment:
Maybe add a comment here explaining why we're doing this in such a verbose way?

Reply:
++, done in 2ebfc6cb33434cf0005780348d01089a4376ee77